File: pkg/api/episode.go
package api

import (
	"github.com/spiceai/spiceai/pkg/flights"
	"github.com/spiceai/spiceai/pkg/proto/runtime_pb"
)

func NewEpisode(ep *flights.Episode) *runtime_pb.Episode {
	return &runtime_pb.Episode{
		Episode:      ep.EpisodeId,
		Start:        ep.Start.Unix(),
		End:          ep.End.Unix(),
		Score:        ep.Score,
		ActionsTaken: ep.ActionsTaken,
		Error:        ep.Error,
		ErrorMessage: ep.ErrorMessage,
	}
}

File: dashboard/src/models/episode.ts
export interface ApiEpisode {
    episode: number,
    start: number,
    end: number,
    score: number,
    actions_taken: Map<string, number>
    error: string
    error_message: string
}

export interface Episode {
    episode: number,
    start: Date,
    end: Date,
    reward: number,
    actions_taken: Map<string, number>
    error: string
    error_message: string
}

export const newEpisode = (apiEpisode: ApiEpisode): Episode =>{
    return {
        episode: apiEpisode.episode,
        start: new Date(apiEpisode.start * 1000),
        end: new Date(apiEpisode.end * 1000),
        reward: apiEpisode.score,
        actions_taken: apiEpisode.actions_taken,
        error: apiEpisode.error,
        error_message: apiEpisode.error_message
    } as Episode
}